src/cmd/product.js

```diff
@@ -20,22 +23,86 @@ module.exports = class ProductCommand {
 		spinnerMixin(this);
 	}
 
+	addDevice({ params: { product, device } }){
+		const identifiers = [device];
+		const msg = `Adding device ${device} to product ${product}`;
+		const upload = uploadProductDevices(product, identifiers);
+		return this.showBusySpinnerUntilResolved(msg, upload)
+			.then(result => this.showDeviceAddResult(result));
+	}
+
+	addDevices({ file, params: { product, device } }){
+		if (!device && !file){
+			throw usageError(
+				'`device` parameter or `--file` option is required'
+			);
+		}
+
+		if (device){
+			return this.addDevice({ params: { product, device } });
+		}
+
+		const msg = `Adding devices in ${file} to product ${product}`;
+		const upload = readDeviceListFile(file).then((identifiers) => uploadProductDevices(product, identifiers));
+		return this.showBusySpinnerUntilResolved(msg, upload)
+			.then(result => this.showDeviceAddResult(result));
+	}
+
+	showDeviceAddResult({ product, identifiers, status: { invalidDeviceIds = [], nonmemberDeviceIds = [] } } = {}){
+		identifiers = filterDeviceIdentifiers(identifiers, invalidDeviceIds, nonmemberDeviceIds);
+		const message = [];
+
+		if (identifiers.length){
+			message.push(
+				'Success!',
+				'',
+				`Product ${product} Includes:`,
+				dedupeAndStringifyIDList(identifiers),
+				''
+			);
+		}
+
+		// TODO (mirande): it seems the API never returns this? currently
+		// adding a device id which is already in another product yields
+		// an api response flagging that id as `invalidDeviceIds`
```

Comment:
Looking at the test that is passing for API service around `nonmemberDeviceIds` it looks like that's returned when a device that is owned by another person who is not part of the org is provided as part of the set.

The filter that builds the `nonmemberDeviceIds` set confirms that:

```js
if (device.user_id || device.customer_id) {
	return device.belongsToOrg(org).then(function returnInvalid(belongs) {
		return !belongs;
	});
}
return false;
```

https://github.com/particle-iot/api-service/blob/master/src/contexts/productDeviceImportContext.js#L142

Reply:
🤔 i still can't get it to return anything in `nonmemberDeviceIds` - here's what i've tried:

my test user has a product, an org and an org product. when i try to add a device that belongs to me to either of my test user's products, i get an "invalid id" response. likewise, when i try to add one of my product devices to either those target products, i get the same error state. lastly, i tried adding a device which belongs to the capture test fleet (an org product) and got the same "invalid id" response.

cc @idokleinman any ideas?